specs/GetFavicon.md

```diff
@@ -0,0 +1,155 @@
+```cpp
+CHECK_FAILURE(m_webView->add_FaviconChanged(
+    Callback<ICoreWebView2FaviconChangedEventHandler>(
+    [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT {
+
+    wil::com_ptr<IStream> iconStream = SHCreateMemStream(nullptr, 0);
+
+    sender->GetFavicon(COREWEBVIEW2_FAVICON_IMAGE_FORMAT_PNG, iconStream.get(),
+        Callback<ICoreWebView2ExperimentalGetFaviconCompletedHandler>(
+            [iconStream, this](HRESULT error_code) -> HRESULT
+            {
+                if (error_code == S_OK)
+                {
+                    Gdiplus::Bitmap* iconBitmap =
+                        new Gdiplus::Bitmap(iconStream.get());
+                    HICON icon;
+                    if (!iconBitmap->GetHICON(&icon))
+                    {
+                        SendMessage(
+                            m_appWindow->GetMainWindow(), WM_SETICON,
+                            ICON_SMALL, (LPARAM)icon);
+                    }
+                }
+
+                return S_OK;
+            }).Get());
+
+            return S_OK;
+    }).Get(), &m_faviconChangedToken));
+```
+## .NET / WinRT Registering a listener for favicon changes
+```c#
+webView.CoreWebView2.FaviconChanged += (CoreWebView2 sender, Object arg) =>
+{
+    System.IO.Stream stream = new System.IO.MemoryStream();
+    await webView.CoreWebView2.GetFaviconAsync(
+        CoreWebView2FaviconImageFormat.Png,
+        stream);
+    // setting the window Icon to the bitmap
+    this.Icon = BitmapFrame.Create(stream); 
+
+};
+```
+# API Notes
+Note that even if a web page does not have a Favicon, the FaviconChanged event
+is raised when the page is navigated to. The Favicon would be an
+empty image stream and empty Uri for the lack of a favicon. The end developer is expected to handle this scenario.
+Otherwise, we raise the FaviconChanged with an observed change to the
+Favicon. In that scenario, the CoreWebView2 has an updated value for the FaviconUri property, and the GetFavicon method to match the updated favicon.
+See [API Details](#api-details) Section below for API reference
+# API Details
+## Win32 C++
+```cpp
+/// This interface is a handler for when the `Favicon` is changed.
+/// The sender is the ICoreWebView2 object the top-level document of 
+/// which has changed favicon and the eventArgs is nullptr. Use the 
+/// FaviconUri property and GetFavicon method to obtain the favicon 
+/// data. The second argument is always null.
+/// For more information see `add_FaviconChanged`.
+[uuid(2913DA94-833D-4DE0-8DCA-900FC524A1A4), object, pointer_default(unique)]
+interface ICoreWebView2FaviconChangedEventHandler : IUnknown {
+  /// Called to notify the favicon changed.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] IUnknown* args);
+}
+
+/// This interface is a handler for the completion of the copying for the`imageStream`.
+/// The 'error_code` is E_NOT_SET if the there is no image. Otherwise error_code 
```

Comment:
Do we have precedent for this use of E_NOT_SET in webview2 APIs?  An E_ code here looks like it would mean an Exception for the WinRT version if I try to get the Favicon on a website that didn't specify one, which doesn't seem exceptional. 
(In other words should this be able to indicate Empty without an error?)

Reply:
I agree that having no image shouldn't be exceptional. We should change this to return S_OK and an empty stream in that case.

Reply:
Remove the error code since the new API will just return an empty stream and there is no data on why.